holoviews/core/element.py

```diff
@@ -378,6 +378,8 @@ class Element3D(Element2D):
 
     __abstract = True
 
+    _selection_streams = ()
```

Comment:
I'm a little concerned about elements having more state attached to them that needs to reasoned about (e.g for pickling?)

Reply:
`_selection_streams` is an immutable class-level property that indicates which stream types can be used on the element to generate selections, so I don't think it should be an issue here for pickling.

Reply:
Looking at [line 296](https://github.com/holoviz/holoviews/blob/26a0afb7c0b58c625b48769644a836c3195a07a3/holoviews/selection.py#L296) of `selection.py` I see the named tuple being built:

```
return _SelectionStreams(
            colors_stream=colors_stream,
            exprs_stream=exprs_stream,
            cmap_streams=cmap_streams,
        )
```

Where I see that `exprs_stream = _Exprs(exprs=[True, None])` and `_Exprs` is defined using `Stream.define`. That would make it an instance of a stream (i.e stateful) as opposed to a just a type (i.e just the stream class). 

If `_selection_streams` really does contains the types and not the instances (which doesn't seem to be the case unless I am misunderstanding things!) then it should be called something like `_selection_stream_types`.

Reply:
It is true that the named tuple is immutable, but the things inside it are not which means it becomes the responsibility of the code using `_selection_streams` not to mutate anything. I haven't checked whether this is the case yet but it could be that these streams are not actually used directly anywhere but are used to create new objects elsewhere....

Reply:
> If _selection_streams really does contains the types and not the instances (which doesn't seem to be the case unless I am misunderstanding things!) then it should be called something like _selection_stream_types.

Yes, it should be named `_selection_stream_types`. I'll make that change.

Reply:
I'm also confused now, as @jlstevens points out doesn't this line make an instance: https://github.com/holoviz/holoviews/blob/26a0afb7c0b58c625b48769644a836c3195a07a3/holoviews/selection.py#L280

Reply:
The named tuple instances created in that `_build_selection_streams` method are only stored by the `link_selections` `Operation` object.  The `Element` class-level tuple currently named `_selection_streams` (which should be named `_selection_stream_types`) is what stores a tuple of `LinkedStream` types, not any of the named tuple types defined in `selection.py`.

Does that help at all?

Reply:
Ah, yes, still digging through the code. Thanks for explaining it. Let me know if you want me to take over the PR since you indicated you won't have more time until later next week.

Reply:
Yes, thanks for explaining! I think it will be fine after the renaming...